modules/apps/transfer/types/msgs.go

```diff
@@ -69,6 +70,10 @@ func (msg MsgTransfer) ValidateBasic() error {
 	if strings.TrimSpace(msg.Receiver) == "" {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "missing recipient address")
 	}
+	receiver := []rune(msg.Receiver)
```

Comment:
is the []rune necessary?

Reply:
I convert to `[]rune` so that `len` gives me the number of characters that the string has. Otherwise, if I do `len` directly on `msg.Receiver`, it gives me the number of bytes of the string, which maybe or may not be the same as the number of characters.

Reply:
But we want to check size right? It'd be preferable to ensure the address does not go above a certain size as opposed to character length?

Reply:
For UTF-8 encoded strings one character could be up to 4 bytes, so if we count character and, in the worst case scenario all of them use 4 bytes, then the max byte length that we would allow would be 4*2048 bytes. I thought of counting characters instead of bytes, because the error message would be maybe nicer for the user: you can tell the user that the string is too long for so many characters, but if you tell the user that the string is too long in number of bytes, then it's harder to know by how many characters is too long... Although it would be strange to have non-alpha numeric characters in the recipient address, and in that case they are encoded with 1 byte... So maybe I am just overthinking this too much and we can just count bytes... :D

Reply:
My preference would be to stick with the normal way we do length validation, which is by counting bytes. We shouldn't need to enforce length validation on the receiver address in the first place